projects/storefrontapp-e2e-cypress/cypress/integration/regression/infinite-scroll.e2e-spec.ts

```diff
@@ -1,50 +1,84 @@
 import {
-  infiniteScrollNoShowMoreTest,
-  infiniteScrollNotActivatedTest,
-  infiniteScrollWithShowMoreAtALimit,
-  infiniteScrollWithShowMoreAtTheBeginningTest,
-  scrollConfig,
-  verifyProductListLoaded,
+  configScroll,
+  isPaginationNotVisible,
+  backtoTopIsNotVisible,
+  scrollToFooter,
+  backToTopIsVisible,
+  isPaginationVisible,
+  verifySortingResetsList,
+  verifyFilterResetsList,
+  verifyGridResetsList,
 } from '../../helpers/infinite-scroll';
+import {
+  createAllProductQuery,
+  QUERY_ALIAS,
+} from '../../helpers/product-search';
+
+describe('Infinite scroll', () => {
+  const testUrl = '/Open-Catalogue/Components/Power-Supplies/c/816';
+  const productQueryAlias = `@${QUERY_ALIAS.INFINITE_SCROLL_PRODUCT_LOADED}`;
 
-context('Infinite scroll', () => {
   before(() => {
     cy.window().then((win) => win.sessionStorage.clear());
   });
 
-  describe('infinite scroll is active and no show more button', () => {
-    beforeEach(() => {
-      scrollConfig(true, 0, false);
-      verifyProductListLoaded();
-    });
-
-    infiniteScrollNoShowMoreTest();
+  beforeEach(() => {
+    cy.server();
+    createAllProductQuery(QUERY_ALIAS.INFINITE_SCROLL_PRODUCT_LOADED);
   });
 
-  describe('infinite scroll should be active using the show more button from the beginning', () => {
-    beforeEach(() => {
-      scrollConfig(true, 0, true);
-      verifyProductListLoaded();
-    });
+  it("should enable Infinite scroll and NOT display 'Show more' button", () => {
```

Comment:
I don't mind having the test like this, but I guess it will be a topic for tomorrow when explaining the best practices, but wouldn't it be best to isolate what is it doing like in it's own `its` for assertions which enables us to find where it failed easier or no?

Reply:
other than that, I believe it's fine after talking about the recommendations if we need to update this test to get # from response then check the breadcrumb if it corresponds

Reply:
^ was thinking about the first point, but scratch what I said. Having it produce the end result from the description is all we want